src/main/scala/com/metl/model/BackendAdaptor.scala

```diff
@@ -550,3 +551,28 @@ class ResourceCachingAdaptor(sc:ServerConfiguration,cacheConfig:CacheConfig) ext
     super.isReady
   }
 }
+
+class D2LCachingAdaptor(cacheConfig:CacheConfig) {
```

Comment:
TLDR:  I'd be inclined to say just put down a cache which is queryTerm=>JValue, and let's add the generation moment to the JValue.  I agree that the tuning isn't important.  Let's put it at something low like 1 minute, and leave it there in code.  We'll move it out to config, if (and only if) we find ourselves changing the expiry time twice after we put it into prod.

no concurrency on foreach unless you specifically get it in using ```List.par.foreach(parFunc).toList```.  I'll talk you through that one over the phone, but it's WAY COOL.  ;)

a) 5s is a long time for prod.  When I have 5s, I'm optimizing, until I can't.  We do have some things that cost high times in prod, and I'm mighty sad about them too.  ;)
b) Agreed - let's keep the expiry time low - 1 hour would be unacceptable.  But, we do need to protect ourselves against somebody mashing the refresh button, especially with touchscreen devices which might have poor visual feedback of button presses, etc.
c) As to the question of how quickly class lists change, surprisingly often.  I wish they only happened once per semester or term, but students can change classes remarkably regularly before census date, and even after census date they may change groups within classes (where groups here refers to a particular session - date&time&place weekly).  So, the classlists do change pretty regularly, and we do see multiple changes per day, but usually the teacher is highly aware of them.  

If we aim for anywhere between 1 and 5 minute cache expiry, I doubt the teacher would mind or even notice.  As long as we put the time of querying on the screen so that they can see it.  My feeling is 1 minute.